src/main/java/org/example/bookshop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+import org.example.bookshop.dto.OrderDto;
+import org.example.bookshop.dto.OrderItemDto;
+import org.example.bookshop.dto.OrderPurchaseRequestDto;
+import org.example.bookshop.dto.OrderStatusRequestDto;
+import org.example.bookshop.exception.DataProcessingException;
+import org.example.bookshop.exception.EntityNotFoundException;
+import org.example.bookshop.mapper.OrderItemMapper;
+import org.example.bookshop.mapper.OrderMapper;
+import org.example.bookshop.model.Order;
+import org.example.bookshop.model.OrderItem;
+import org.example.bookshop.model.ShoppingCart;
+import org.example.bookshop.model.User;
+import org.example.bookshop.repository.CartItemRepository;
+import org.example.bookshop.repository.CartRepository;
+import org.example.bookshop.repository.OrderItemRepository;
+import org.example.bookshop.repository.OrderRepository;
+import org.example.bookshop.service.OrderService;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final CartItemRepository cartItemRepository;
+    private final CartRepository cartRepository;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderMapper orderMapper;
+
+    @Override
+    @Transactional
+    public OrderDto add(OrderPurchaseRequestDto requestDto, User user) {
+        ShoppingCart shoppingCart = cartRepository.findByUserId(
+                user.getId()).orElseThrow(() -> new EntityNotFoundException(
+                "Can't find order for user: " + user.getEmail()));
+        Order order = createOrder(shoppingCart, requestDto, user);
+        shoppingCart.getCartItems().clear();
+        cartRepository.save(shoppingCart);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public Page<OrderDto> getAll(User user, Sort sort, Pageable pageable) {
+        return orderRepository.findAllByUserId(user.getId(), pageable)
+                .map(orderMapper::toDto);
+    }
+
+    @Override
+    public void update(OrderStatusRequestDto requestDto, Long orderId, User user) {
+        Order order = findOrderByIdAndUserId(orderId, user.getId());
+        Order.Status status = Order.Status.valueOf(requestDto.getStatus());
+        if (order.getStatus().equals(status)) {
+            throw new DataProcessingException("Order already has status: " + status);
+        }
+        order.setStatus(status);
+        orderRepository.save(order);
+    }
+
+    @Override
+    public Set<OrderItemDto> get(Long orderId, User user) {
+        findOrderByIdAndUserId(orderId, user.getId());
+        return orderItemRepository.findAllByOrderId(orderId).stream()
```

Comment:
Don't need to request DB twice, 
see an example
```suggestion
        Order order = orderRepository.findByIdAndUserId(orderId, userId)....
        return orderItemMapper.toOrderItemDtoList(order.getOrderItems());
```